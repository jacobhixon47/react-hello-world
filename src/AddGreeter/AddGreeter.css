.add-greeter {
  margin: 20px;
  padding: 20px;
  width: 100%;
  border: 2px solid #00FFD8;
  background: #DDFFEE;
  text-align: center;
}

input {
  padding: 1%;
  font-size: 1.5em;
}
