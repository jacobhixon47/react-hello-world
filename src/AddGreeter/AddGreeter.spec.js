import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddGreeter from './AddGreeter';

describe(AddGreeter, () => {
  // mock function
  const mockAddGreeting = jest.fn();
  // create shallow component
  const component = shallow(
    <AddGreeter addGreeting={mockAddGreeting}/>
  );
  // snapshot test
  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <AddGreeter />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  // test that form exists
  it('contains the add-greeting form', () => {
    expect(component.find('input')).toHaveLength(1);
    expect(component.find('button')).toHaveLength(1);
  });
  // test addGreeting() functionality
  it('calls addGreeting when add-greeting button is clicked', () => {
    component.find('button').simulate('click');
    expect(mockAddGreeting).toBeCalled();
  });
  // test handleUpdate() functionality
  it('updates greetingName from the form when keys are pressed', () => {
    const updateKey = 'foo';
    component.instance().handleUpdate({ target: { value: updateKey }});
    expect(component.state('greetingName')).toEqual(updateKey);
  });
  // test form reset when 'add' button is clicked
  it('blanks out the greetingName when the button is clicked', () => {
    const updateKey = 'bar';
    component.instance().handleUpdate({ target: { value: updateKey }});
    expect(component.state('greetingName')).toEqual(updateKey);
    component.find('button').simulate('click');
    expect(component.state('greetingName')).toHaveLength(0);
  });
});
