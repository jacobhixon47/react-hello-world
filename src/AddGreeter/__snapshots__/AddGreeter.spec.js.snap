// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class AddGreeter extends _react.Component {
  constructor(props) {
    super(props);
    this.state = { greetingName: '' };
    this.handleUpdate = this.handleUpdate.bind(this);
    this.addGreeting = this.addGreeting.bind(this);
  }
  addGreeting() {
    this.props.addGreeting(this.state.greetingName);
    this.setState({ greetingName: '' });
  }
  handleUpdate(event) {
    this.setState({ greetingName: event.target.value });
  }
  render() {
    return (
      _react2.default.createElement('div', { className: 'add-greeter', __source: { fileName: _jsxFileName, lineNumber: 20 }, __self: this },
        _react2.default.createElement('input', {
          type: 'text',
          onChange: this.handleUpdate,
          value: this.state.greetingName, __source: { fileName: _jsxFileName, lineNumber: 21 }, __self: this }),

        _react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 26 }, __self: this }),
        _react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 27 }, __self: this }),
        _react2.default.createElement('button', { onClick: this.addGreeting, __source: { fileName: _jsxFileName, lineNumber: 28 }, __self: this }, 'Add')));


  }} renders and matches our snapshot 1`] = `
<div
  className="add-greeter"
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
